### PR TITLE
fix: (platform) Removed Invalid examples and compact menu button fix

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3357
https://github.com/SAP/fundamental-ngx/issues/3364

#### Please provide a brief summary of this pull request.
This PR fixes both issues.
1. Remove invalid example fro menu button.
2. Include compact density for menu and modify compact menu example for the same.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by sKudum — APPROVED** [review]
LGTM

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Review by InnaAtanasova — APPROVED** [review]

**Review by droshev — DISMISSED** [review]
The second issue is still not solved - the compact menu button has cozy menu
Screen Shot 2020-09-24 at 8 31 41 PM

**Comment by DeepakSap14** [comment]
> The second issue is still not solved - the compact menu button has cozy menu
> Screen Shot 2020-09-24 at 8 31 41 PM

@droshev for this i have raised separate issue to track. https://github.com/SAP/fundamental-ngx/issues/3364
It relates to Platform menu, currently platform menu is not supporting compact density.

**Comment by DeepakSap14** [comment]
I have included fix for Menu compact density. and modified example for compact menu button.

File: libs/platform/src/lib/components/menu/menu.component.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <ng-template #menuTemplate>
     <div class="fd-popover__body fd-popover__body--no-arrow" [attr.id]="menuId" [attr.dir]="direction">
-        <nav class="fd-menu" tabindex="0" role="menu" (click)="onClick($event)" (keydown)="onKeydown($event)">
+        <nav class="fd-menu" [ngClass]="contentDensity==='compact'? 'fd-menu--compact': ''" tabindex="0" role="menu"
```
Comment (Betrozov) [inline libs/platform/src/lib/components/menu/menu.component.html:3]: plz use `[class.fd-menu--compact]="contentDensity === 'compact'"`

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/menu/menu.component.html (hunk 1 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/menu/menu.component.html]: Done

**Review by KevinOkamoto — APPROVED** [review]
Looks good!

**Comment by DeepakSap14** [comment]
will wait till https://github.com/SAP/fundamental-ngx/pull/3400/ is merged.
Have to implement breaking changes.

File: libs/platform/src/lib/components/menu/menu.component.ts [hunk 2]
```diff
@@ -68,6 +69,10 @@ export class MenuComponent implements OnInit, AfterViewInit, AfterContentInit, O
         this.menuId = MENU_ID_ROOT + id;
     }
 
+    /** Density for Menu */
+    @Input()
+    contentDensity: ContentDensity
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/menu/menu.component.ts:74]: Should not we have a default value?

Comment (DeepakSap14) [reply libs/platform/src/lib/components/menu/menu.component.ts]: fixed. included default as 'cozy'.

**Comment by DeepakSap14** [comment]
This can be reviewed now.
@JKMarkowski

**Comment by DeepakSap14** [comment]
@dimamarksman

**Review by dimamarksman — APPROVED** [review]
LGTM

**Review by JKMarkowski — APPROVED** [review]

**Pull request merged**